src/requirements/requirement-frontend-utils.ts

```diff
@@ -501,9 +527,6 @@ export function getRelatedUnfulfilledRequirements(
         toggleableRequirementChoices
       );
       // potential self-check requirements
-      if (requirementSpec == null && !subRequirement.allowCourseDoubleCounting) {
```

Comment:
Why do we not need the second check here anymore?

Reply:
The function is only called once in `NewCourseModal.vue`, to generate the automatically fulfilled requirements and the non-automatically fulfilled requirements dropdown. In the past, the if-expression was used to filter out the requirements that allowed double counting which we assumed would always be automatically fulfilled. However, with our reframing of the double counting flag (and in this new implementation), it is filtered later in a similar way.